source/text_handler.cpp

```diff
@@ -229,25 +229,24 @@ spv_result_t AssemblyContext::binaryEncodeU64(const uint64_t value,
 spv_result_t AssemblyContext::binaryEncodeNumericLiteral(
     const char* val, spv_result_t error_code, const IdType& type,
     spv_instruction_t* pInst) {
-  const bool is_bottom = type.type_class == libspirv::IdTypeClass::kBottom;
-  const bool is_floating = libspirv::isScalarFloating(type);
-  const bool is_integer = libspirv::isScalarIntegral(type);
-
-  if (!is_bottom && !is_floating && !is_integer) {
-    return diagnostic(SPV_ERROR_INTERNAL)
-           << "The expected type is not a scalar integer or float type";
+  using spvutils::EncodeNumberStatus;
+  EncodeNumberStatus rc = EncodeNumberStatus::kSuccess;
```

Comment:
Fixed in 654395d.

Hmm, but the type is EncodeNumberStatus, and the variable name is parse_status, looks a little bit inconsistent.